server/src-lib/Hasura/Server/PGDump.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Hasura.Server.PGDump
+  ( PGDumpReqBody
+  , execPGDump
+  ) where
+
+import           Data.Aeson.Casing
+import           Data.Aeson.TH
+import qualified Data.ByteString.Lazy   as BL
+import qualified Data.FileEmbed         as FE
+import qualified Data.List              as L
+import qualified Data.Text              as T
+import qualified Database.PG.Query      as Q
+import           Hasura.Prelude
+import qualified Hasura.RQL.Types.Error as RTE
+import           System.Exit
+import           System.Process
+
+data PGDumpReqBody =
+  PGDumpReqBody
+  { prbOpts        :: ![String]
+  , prbCleanOutput :: !(Maybe Bool)
+  } deriving (Show, Eq)
+
+$(deriveJSON (aesonDrop 3 snakeCase) ''PGDumpReqBody)
+
+script :: IsString a => a
+script = $(FE.embedStringFile "src-rsr/run_pg_dump.sh")
+
+runScript
+  :: String
+  -> [String]
+  -> String
+  -> IO (Either String BL.ByteString)
+runScript dbUrl opts clean = do
+  (exitCode, filename, stdErr) <- readProcessWithExitCode "/bin/sh"
+    ["/dev/stdin", dbUrl, unwords opts, clean] script
+  case exitCode of
+    ExitSuccess   -> do
+      contents <- BL.readFile $ L.dropWhileEnd (== '\n') filename
+      return $ Right contents
+    ExitFailure _ ->  return $ Left stdErr
+
+execPGDump
+  :: (MonadError RTE.QErr m, MonadIO m)
+  => PGDumpReqBody
+  -> Q.ConnInfo
+  -> m BL.ByteString
+execPGDump b ci = do
+  output <- liftIO $ runScript dbUrl opts clean
```

Comment:
`runScript` is prone to IO exception when the `filename` is not found in disk or maybe other reasons. Use `try` to catch IO exception explicitly.